### DISCUSSION TITLE
Metric counting number of GitHub Pages deployments

### DISCUSSION THREAD

**Discussion by spenserblack** [lowlighter/metrics#423] [Ideas]
If possible/reasonable, this might be a fun metric to have.

**Comment by lowlighter** [comment]
Mainly a few questions about how this would work:
- Would it be only for GitHub pages, or all environment ?
- Does it count the total active deployment, or are all deployments (including redeployment in these) ?

Of course, design can be limited by GitHub api too, I didn't check yet what it's possible to do with it for deployments 🙂

**Reply by spenserblack** [reply]
>Of course, design can be limited by GitHub api too

Yeah, I should have checked if the API supports it. Sorry, I was a bit too hasty in posting this idea after it popped into my head 😅 

I was originally thinking only GitHub pages, active deployments.

But I'm kind of changing my mind on what the metric should be as I type this:
I believe the API has a nullable "website" value for repos. That changes from a deployment count metric to a website count metric, which might be better.

**Reply by lowlighter** [reply]
> Yeah, I should have checked if the API supports it. Sorry, I was a bit too hasty in posting this idea after it popped into my head 😅

No worries, even if ideas can not be implemented because of technical implementations, it's still useful for future ideas or in case GitHub API evolves in the future 🙂 

> I believe the API has a nullable "website" value for repos. That changes from a deployment count metric to a website count metric, which might be better.

Indeed, it's the `homepageUrl`, though it can be any website, not necessarly a GitHub pages. I did not find any direct reference to check whether GitHub pages are enabled.

However it seems possible to track deployments and environments, but it seems weird to consider them as user metrics. Could be used when using metrics for repositories though. 

```
{
  "data": {
    "repository": {
      "deployments": {
        "totalCount": 435
      },
      "environments": {
        "totalCount": 2
      },
      "homepageUrl": "https://metrics.lecoq.io",
      "projectsUrl": "https://github.com/lowlighter/metrics/projects"
    }
  }
}
```

**Reply by spenserblack** [reply]
>...but it seems weird to consider them as user metrics. Could be used when using metrics for repositories though.

Yeah, as a repository metric it sounds cool!

For users, perhaps it could be an achievement, in the same spirit as the "Automater" achievment :thinking:
